org.civicrm.reqangular/src/common/services/api/settings.api.js

```diff
@@ -0,0 +1,33 @@
+/* eslint-env amd */
+
+define([
+  'common/lodash',
+  'common/modules/apis',
+  'common/services/api'
+], function (_, apis) {
+  'use strict';
+
+  apis.factory('api.settings', ['$log', 'api', function ($log, api) {
+    $log.debug('api.settings');
+
+    return api.extend({
+
+      /**
+       * Returns all settings.
+       *
+       * @param  {Object} params  matches the api endpoint params (ex. contact_id)
+       * @return {Promise}
```

Comment:
Please add what the promise resolves with

Reply:
sure! added!